packages/multi-test/src/contracts.rs

```diff
@@ -337,8 +337,20 @@ where
         info: MessageInfo,
         msg: Vec<u8>,
     ) -> AnyResult<Response<C>> {
-        let msg = from_slice(&msg)?;
-        (self.execute_fn)(deps, env, info, msg).map_err(|err| anyhow!(err))
+        let msg: T1 = from_slice(&msg)?;
+        let address = env.contract.address.clone();
+        (self.execute_fn)(deps, env, info.clone(), msg.clone())
+            .map_err(|err| anyhow!("{}", err))
+            .context(format!(
```

Comment:
@ueco-jb This stuff takes the `ContractError` you'd normally return in a contract entrypoint, makes an `anyhow::Error` out of it, and then wraps that in another `anyhow::Error`, now with extra info (address, sender, etc). So we're not getting extra stdout logs, it's just the error now has more info.

If you look at the failing test, it refers to places like this in code: https://github.com/CosmWasm/cw-plus/blob/main/contracts/cw3-flex-multisig/src/contract.rs#L1089. Basically it looks like `downcast` stopped working with the added context, and I'm not sure why yet.

If you're happy with the direction though, I'll investigate when I'm back.

Reply:
Adding context to the error does seem like a very nice way of handling it.

I assume this context is print out when we do `res.unwrap()` on the `Err` variant. If so, this seems like a very nice addition.

Question: if this is nested - execute calls execute in submsg, 2nd contract errors - this will show context for both contracts, right?

A test showing the error message in such a case would be great to evaluate it. (I know... super hard to do CI there, maybe just something marked "#[skip]" that I can modify that line and run locally to evaluate

Reply:
> Question: if this is nested - execute calls execute in submsg, 2nd contract errors - this will show context for both contracts, right?

Yes, although I haven't tested it with more levels of nesting I think. A test would be nice for that. I'll look at that.

Reply:
Okay, the nested thing didn't quite work, but I fixed it. There are tests showing it works. There's a way to take a peek what the display looks like now: https://github.com/CosmWasm/cw-plus/pull/597/files#r778142632